backend/cmmgen.ml

```diff
@@ -398,12 +398,18 @@ let exttype_of_sort (s : Layouts.Sort.const) =
   match s with
   | Value -> XInt
   | Float64 -> XFloat
+  | Word -> XInt
+  | Bits32 -> XInt32
+  | Bits64 -> XInt64
   | Void -> Misc.fatal_error "Cmmgen.exttype_of_sort: void encountered"
 
 let machtype_of_sort (s : Layouts.Sort.const) =
   match s with
   | Value -> typ_val
   | Float64 -> typ_float
+  | Word -> typ_int
```

Comment:
I'm not sure about this.  `Word` is not GC-scannable, right?

Reply:
`word` is not GC-scannable. It’s intended to be word-sized (64 bits sometimes and 32 bits sometimes). It’s the layout for `nativeint#`. Of course, we don’t actually support any 32-bit native backends at the moment, so it’s a bit academic, but I think that will change.

Reply:
Hmm it should probably be one of the new `typ_int32` or `typ_int64` then.  However now that I think about it, I'm unsure we should be introducing either of these.  I need to consider this some more.

Reply:
I now think we should just use `typ_int` for these, as discussed internally at JS.